### ISSUE TITLE
Drop Down Arrow not visible

### ISSUE THREAD

**Issue by ravivlb** [jriosdev/iOSDropDown#8]
#4 

I am still facing issue in iOS 12. 

Can you share way to show arrow in 12.

**Comment by jriosdev** [comment]
iOS12 is Still beta, that's why I didn't test on it

**Comment by tmrbaset** [comment]
You need to change the self.layer.mask method and better make it self.layer.addSublayer in the Arrow draw function


```
override func draw(_ rect: CGRect) {
        
        // Get size
        let size = self.layer.frame.width
        
        // Create path
        let bezierPath = UIBezierPath()
        
        // Draw points
        let qSize = size/4
        
        bezierPath.move(to: CGPoint(x: 0, y: qSize))
        bezierPath.addLine(to: CGPoint(x: size, y: qSize))
        bezierPath.addLine(to: CGPoint(x: size/2, y: qSize*3))
        bezierPath.addLine(to: CGPoint(x: 0, y: qSize))
        bezierPath.close()
        
        // Mask to path
        let shapeLayer = CAShapeLayer()
        shapeLayer.path = bezierPath.cgPath

       // Add a color, can be added as property
       shapeLayer.fillColor = UIColor.white.cgColor

     // Make the change here
     self.layer.addSublayer (shapeLayer)


    }
```

**Comment by jriosdev** [comment]
thanks for your contribution

**Comment by iosdroid** [comment]
Hi

> You need to change the self.layer.mask method and better make it self.layer.addSublayer in the Arrow draw function
> 
> ```
> override func draw(_ rect: CGRect) {
>         
>         // Get size
>         let size = self.layer.frame.width
>         
>         // Create path
>         let bezierPath = UIBezierPath()
>         
>         // Draw points
>         let qSize = size/4
>         
>         bezierPath.move(to: CGPoint(x: 0, y: qSize))
>         bezierPath.addLine(to: CGPoint(x: size, y: qSize))
>         bezierPath.addLine(to: CGPoint(x: size/2, y: qSize*3))
>         bezierPath.addLine(to: CGPoint(x: 0, y: qSize))
>         bezierPath.close()
>         
>         // Mask to path
>         let shapeLayer = CAShapeLayer()
>         shapeLayer.path = bezierPath.cgPath
> 
>        // Add a color, can be added as property
>        shapeLayer.fillColor = UIColor.white.cgColor
> 
>      // Make the change here
>      self.layer.addSublayer (shapeLayer)
> 
> 
>     }
> ```

But This is not working on ios 11

**Comment by iosdroid** [comment]
> shapeLayer.fillColor = UIColor.white.cgColor

override func draw(_ rect: CGRect) {
        
        // Get size
        let size = self.layer.frame.width
        
        // Create path
        let bezierPath = UIBezierPath()
        
        // Draw points
        let qSize = size/4
        
        bezierPath.move(to: CGPoint(x: 0, y: qSize))
        bezierPath.addLine(to: CGPoint(x: size, y: qSize))
        bezierPath.addLine(to: CGPoint(x: size/2, y: qSize*3))
        bezierPath.addLine(to: CGPoint(x: 0, y: qSize))
        bezierPath.close()
        
        // Mask to path
        let shapeLayer = CAShapeLayer()
        shapeLayer.path = bezierPath.cgPath
        // Add a color, can be added as property
        shapeLayer.fillColor = UIColor.black.cgColor
        
        if #available(iOS 12.0, *) {
            // Make the change here
            self.layer.addSublayer (shapeLayer)
        } else {
             self.layer.mask = shapeLayer
        }
      
        
        
        
        
    }

**Comment by jriosdev** [comment]
thank you all
